services/httpd/handler.go

```diff
@@ -1087,13 +1089,23 @@ func gzipFilter(inner http.Handler) http.Handler {
 			inner.ServeHTTP(w, r)
 			return
 		}
-		gz := gzip.NewWriter(w)
-		defer gz.Close()
+		gz := gzipWriterPool.Get().(*gzip.Writer)
+		gz.Reset(w)
+		defer func() {
+			gz.Close()
+			gzipWriterPool.Put(gz)
```

Comment:
If you want I will add it, but it should be redundant... objects in pools are discarded during GC, so any reference they hold to other objects won't keep the other objects alive. And for being in the same state, it shouldn't really matter because we call `gz.Reset(w)` in any case.